web/src/compose_setup.js

```diff
@@ -179,9 +181,16 @@ export function initialize() {
                     // user-triggered edit to that field.
                     $input.trigger("input");
 
-                    // TODO: Probably this should also renarrow to the
-                    // new topic, if we were currently viewing the old
-                    // topic, just as if a message edit had occurred.
+                    // renarrow to the new topic if currently viewing the old topic
+                    const terms = hash_util.parse_narrow(window.location.hash.split(/\//));
+                    if (terms.length < 2) {
+                        return;
+                    }
+                    const stream_name = sub_store.maybe_get_stream_name(stream_id);
+                    if (terms[0].operand === stream_name && terms[1].operand === topic_name) {
+                        const url = hash_util.by_stream_topic_url(stream_id, new_topic);
+                        browser_history.go_to_location(url);
+                    }
```

Comment:
This should probably use one of the `narrow.by*` methods.

Reply:
I've switched to using `narrow.activate`.